org-roam-node.el

```diff
@@ -829,23 +829,50 @@ Any top level properties drawers are incorporated into the new heading."
     (org-roam-erase-keyword "title")
     (org-roam-erase-keyword "filetags")))
 
+(defun org-roam--heading-levels ()
+  "Extract the heading level structure from the current file."
+  (org-map-entries (lambda () (org-current-level))
+                   nil 'file))
+
+(defun org-roam--can-promote-p ()
+  "Verify that this buffer is promoteable:
+There is a single level-1 heading
+and no extra content before the first heading."
+  (let ((level-1-headings-count (-count (lambda (level) (eq 1 level)) (org-roam--heading-levels)))
```

Comment:
let's not use anything from `dash` here, we have no explicit dependency on `dash` at the moment, although we do, I think, pull it in transitively.